### ISSUE TITLE
How to Paginate?

### ISSUE THREAD

**Issue by jl-DaDar** [pynamodb/PynamoDB#248]
I can't seem to find anyways to obtain the last_evaluated_key in the high level api.
Looking at the Model code, the last_evaluated_key is never passed back to the caller of Model scan, query, or rate_limited_scan. And as a result, I can't do pagination. How are people doing manual pagination with the high level api? Isn't this needed for most applications, especially mobile app? I am so confused.

Thanks.

**Comment by yedpodtrzitko** [comment]
There's a Google Group dedicated to asking questions:
https://groups.google.com/forum/#!forum/pynamodb

Here are a few issues asking exactly the same question:
https://github.com/jlafon/PynamoDB/issues/231
https://github.com/jlafon/PynamoDB/issues/207

And here is a PR trying to implement it:
https://github.com/jlafon/PynamoDB/pull/226


So to answer your question: you have to implement it yourself.

**Comment by 3dw1np** [comment]
Can we except an official implementation to get access to the last evaluted key value ?

**Comment by jl-DaDar** [comment]
a quick solution is to simply reconstruct a last key base on the last item fetched.
and use that directly. Its a dict. But it doesn't seem to work on index query, I tried to reconstruct one base one the index keys but it does not work on the index query. I can only get default keys to works with scan

**Comment by soaxelbrooke** [comment]
For anyone who happens upon this looking for an answer, last_evaluated_key is of the form:

```
last_evaluated_key={'my_hash_key': {'S': 'afsdfasdfasdf'}, 'my_range_key': {'S': 'afsdfasvdfvxz'}}
```

Assuming your hash key and range key are both strings.

**Comment by richiverse** [comment]
This doesn't work for hash keys in global secondary indexes

**Comment by jpinner-lyft** [comment]
Fixed in #379
